### PR TITLE
Feature/red 232 integrated editor

### PR DESCRIPTION
**Pull request by red-tt-bear**

### CODE REVIEW

File: src/utilities/renderer-helper.ts [hunk 1]
```diff
@@ -131,29 +138,29 @@ export const rendererResponseValidationScheme = Joi.object({
     })).required(),
     debug: Joi.object({
         // TODO are these required or optional
-        debug: Joi.array().items(Joi.string()).required(),
-        internal: Joi.array().items(Joi.string()).required(),
-        perl_warn: Joi.string().allow('').required(),
-        pg_warn: Joi.array().items(Joi.string()).required(),
-        render_warn: Joi.array().items(Joi.string()).optional(), // THIS FIELD IS NEW, replace with required 
+        // debug: Joi.array().items(Joi.string()).required(),
```
Comment (red-tt-bear) [inline src/utilities/renderer-helper.ts:141]: @doombot-exe we should sync on this, I had issues where it said it wasn't an array when I used the preview with the source

(but I don't think this happened with the renderer helper)

File: src/features/courses/course-route.ts [hunk 2]
```diff
@@ -1117,4 +1118,67 @@ router.delete('/attachments/:id',
...
+        if (_.isNil(req.session)) {
+            throw new Error(Constants.ErrorMessage.NIL_SESSION_MESSAGE);
+        }
+
+        const basePath = `private/my/${req.session.userId}`;
+        const result = await rendererHelper.catalog({
+            basePath: basePath,
+            // TODO what should the depth be?
+            // -1 did not do all
+            maxDepth: 100,
```
Comment (red-tt-bear) [inline src/features/courses/course-route.ts:1177]: I thought -1 was supposed to do the whole catalog

Comment (drdrew42) [reply src/features/courses/course-route.ts:1177]: I don't think I ever built that in. There is no "unlimited" search.

Comment (PanopticaRising) [reply src/features/courses/course-route.ts:1177]: Should maxDepth be passed in from the UI?

**Review by PanopticaRising — APPROVED** [review]

File: src/features/courses/course-route.ts (hunk 2 above)
Comment (red-tt-bear) [reply src/features/courses/course-route.ts:1177]: In the future yes, the base path prefix is also hardcoded here
I feel like this makes it a little more secure in the meantime

**Pull request merged**